vocabularies/UI.xml

```diff
@@ -1142,8 +1199,31 @@ The trend is
 
       <ComplexType Name="SelectOptionType">
         <Annotation Term="Core.Description" String="List of value ranges for a single property" />
-        <Property Name="PropertyName" Type="Edm.PropertyPath" Nullable="false">
+        <Annotation Term="Core.LongDescription" String="Exactly one of `PropertyName` and `DynamicPropertyName` must be present" />
+        <Property Name="PropertyName" Type="Edm.PropertyPath">
+          <Annotation Term="Common.PrimitivePropertyPath"/>
           <Annotation Term="Core.Description" String="Path to the property" />
+          <Annotation Term="Core.Revisions">
+            <Collection>
+              <Record>
+                <PropertyValue Property="Kind" EnumMember="Core.RevisionKind/Modified"/>
+                <PropertyValue Property="Description" String="Now nullable if `DynamicPropertyName` is present"/>
+              </Record>
+            </Collection>
+          </Annotation>
+        </Property>
+        <Property Name="DynamicPropertyName" Type="Edm.AnnotationPath">
+          <Annotation Term="Core.Description" String="Dynamic property introduced by annotations for which value ranges are specified" />
```

Comment:
How would filtering work on dynamic properties? As far as I know our services don't (fully) support dynamic properties.

Reply:
`$apply=groupby(...,aggregate(... as dynprop))/filter(dynprop lt 1000)`

Reply:
An example would be helpful here: which combination of annotations will trigger the client to create this non-trivial request?

Reply:
extended example accordingly